backend.rb

```diff
@@ -57,14 +56,13 @@ class Backend < Sinatra::Base
       origin_proof:  params[:origin_proof]
     )
 
-    authorization_response = purchase.authorize!
-    if authorization_response.success?
+    success, response_data, response_code = purchase.order!
```

Comment:
I think that the code is number so string matching is wired(not clear), how about < 300?